
error 1: FATAL FAULT `precondition`: debug: (numeric.this **! other)
Call stack:
fuzion.type.runtime.type.fault.type.install_default.λ.call#1: $MODULE/fuzion/runtime/fault.fz:38:7:
      fuzion.sys.fatal_fault kind msg
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fuzion.runtime.fault.cause#1: $MODULE/eff/fallible.fz:35:6:
  => h e
-----^
fuzion.type.runtime.type.pre_fault.type.install_default.λ.call#1: $MODULE/fuzion/runtime/pre_fault.fz:39:7:
      fuzion.runtime.fault.cause ("precondition", msg)).default
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fuzion.runtime.pre_fault.cause#1: $MODULE/eff/fallible.fz:35:6:
  => h e
-----^
fuzion.runtime.precondition_fault#1: $MODULE/fuzion/runtime/pre_fault.fz:58:42:
public precondition_fault(msg String) => pre_fault.cause msg
-----------------------------------------^^^^^^^^^^^^^^^^^^^
i32.pre infix **: $MODULE/numeric.fz:78:7:
      debug: (numeric.this **! other)
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
i32.precall infix **: $MODULE/numeric.fz:77:5:
    pre
----^^^
      debug: (numeric.this **! other)
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      debug: (other ≥ numeric.this.zero)
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
idiom32ex2.exp#3 i32: -:3:41:
  exp     (T type : integer, x, n T) => x **  n
----------------------------------------^^^^^^^
idiom32ex2.loop.loop: -:19:26:
      say "exp $i $j is {exp i j}"
-------------------------^^^^^^^
idiom32ex2.loop: -:18:5:
    for j in 0..15 do
----^
idiom32ex2: -:17:3:
  for i in 0..15 do
--^

*** fatal errors encountered, stopping.
one error.
